lib/Sema/CSSimplify.cpp

```diff
@@ -1568,8 +1568,7 @@ shouldOpenExistentialCallArgument(
       callee, genericSig, genericParam,
       /*treatNonResultCovarianceAsInvariant=*/false,
       /*skipParamIdx=*/paramIdx);
-  if (referenceInfo.selfRef > TypePosition::Covariant ||
-      referenceInfo.assocTypeRef > TypePosition::Covariant)
+  if (referenceInfo.selfRef > TypePosition::Covariant)
```

Comment:
What does this change do?

Reply:
This is to fix opening constrained existential with opaque result type: 

```
 func foo<T>(_ c: some Collection<T>) -> some Collection<T> {
  return c
}
let c: any Collection<Int> = [1, 2, 3] 
let result = foo(c) // causes Type 'any Collection<Int>' cannot conform to 'Collection' error
```

I thought this would fix my failing test but looks like this is a separate issue. Going to make a new PR  for this 🙂